### PR TITLE
Stat: Add panel option to control wide layout

### LINKED ISSUE #77092: Stat: Text is not scaling correctly
### What happened?

After updating from 10.1.5 to 10.2.0 the Stat Panel text does not scale correctly. The "Alias" is way to small to read, normally the Alias was shown on the left and the value on the right in one line.
The settings are all set to "Auto" like on 10.1.5, no changes are made on the settings.

SCR-20231024-thpo

SCR-20231024-tirh

### What did you expect to happen?

Correct alignment of text in the Stat Panel, row is small for this big text... 
Alias and Value should be displayed on one line (left Alias, right Value)

No way to workaround this!

### Did this work before?

Yes, on 10.1.5

### How do we reproduce it?

Update to 10.2.0

### Is the bug inside a dashboard panel?

Yes, attached:
[ENERGIEÜBERWACHUNG OG-1698179934755.json](https://github.com/grafana/grafana/files/13127192/ENERGIEUBERWACHUNG.OG-1698179934755.json)

### Environment (with versions)?

Grafana: 10.2.0
OS: Ubuntu Focal
Browser: Safari

### Grafana platform?

A package manager (APT, YUM, BREW, etc.)

### Datasource(s)?

InfluxDB

### PR DESCRIPTION
**Pull request by nmarrs**
This PR explores adding a panel editor option for stat panel to give users a way back to old "wide layout" functionality of the stat panel.

For context, we recently removed the wide layout functionality based on user feedback. See [this PR](https://github.com/grafana/grafana/pull/75556).

Doing so resulted in backlash from other users as their existing panel designs broke.

This PR hopes to find the middle ground - adding a new panel editor option to the stat panel that allows user to explicitly control whether or not they want wide layout enabled. Adding another panel editor option isn't super ideal, though this option only is enabled when `text mode` is set to `Value and name`.

A key question to consider here is what the default value of `wideLayout` should be (`true` vs `false`). Either way users from one camp or the other will need to make manual changes to their stat panels to fix their designs. IMO given the majority of stat panels created at this point have been with the wide layout functionality, a default value of `true` makes the most sense.

TODO (once initial approach is verified)

- [x] Rethink naming
- [x] Add doc changes

https://github.com/grafana/grafana/assets/22381771/8aa8ff96-fc90-44f7-8597-1ee8a24b40ed

Orientation behavior (no impact)

https://github.com/grafana/grafana/assets/22381771/6554247c-1957-4817-b9f2-8b7b5c77ff87

Stat panel test dashboard
[Stat!-1695827506475.json.txt](https://github.com/grafana/grafana/files/12740282/Stat.-1695827506475.json.txt)

Fixes https://github.com/grafana/support-escalations/issues/7886
Fixes https://github.com/grafana/grafana/issues/77092

### CODE REVIEW

**Comment by bob454522** [comment]
Im your guy! (re: "someone relying on current behavior and that might surprise them")

We have 4x large 4k Monitors showing beautiful grafana dashboards only 24/7 - with panels using lots of stat Visualizations.  Thus was quite surprised after the 10.1.5 to 10.2 update to see alot of our panels overlapping and not looking so good now.

I see there is the option:
        disableWideLayout={true}

**Is this something we can change in Grafana INI?**  Or Is that something internal only where y'all are waiting to potentially add this in the future into the webGUI?      please lmk, as i would like to revert back to wideLayout, If but not sure where to make this change (if it's even possible currently)

bc its worth noting That I think this should not have gone live until the on/off toggle feature was in place (somewhere), partly because by the time users wait for the current WIP to be pushed through and released, most of them will have spent a huge chunk of time rearranging the dashboards that would need this feature.  (As users aren't going to want to have dashboards looking all funky waiting a month or longer for the next release)

tks,
 LOVE GRAFANA!!

>This looks good Nathan! It does look good but I would agree that an option makes sense. I wouldn't be surprised if there is >someone relying on current behavior and that might surprise them. That being said, I think this is still good for now 👍

**Comment by improbable-tomclapham** [comment]
Heya, just in case it's relevant: I noticed that all the examples in this PR and #75556 are of single stats. Our problem case involves multiple stats, in a space too small for a proper table. Here's how it looks now:

multi_stats_panel

This PR's solution looks like it'll be fine for us - much appreciated! I'm just mentioning this as a test case worth watching.

**Review by imatwawana — APPROVED** [review]
Just a couple wording nits, but otherwise looks good to me!

**Comment by drew08t** [comment]
Testing it locally now.

File: public/app/plugins/panel/stat/module.tsx [hunk 1]
```diff
@@ -17,22 +17,36 @@ export const plugin = new PanelPlugin<Options>(StatPanel)
...
-          { value: BigValueTextMode.None, label: 'None' },
-        ],
-      },
-      defaultValue: defaultOptions.textMode,
-    });
+    builder
+      .addSelect({
+        path: 'textMode',
+        name: 'Text mode',
+        description: 'Control if name and value is displayed or just name',
```
Comment (drew08t) [inline public/app/plugins/panel/stat/module.tsx:24]:
```suggestion
        description: 'Control if value, name, or both are displayed',
```

Comment (nmarrs) [reply public/app/plugins/panel/stat/module.tsx:24]: This suggestion is a different meaning 🤔 `textMode` only applies for name + value, name, but not just value 😬

**Review by drew08t — APPROVED** [review]
Other than the minor suggestion regarding the description, LGTM!

**Pull request merged**